### ISSUE TITLE
 Eclipse doesen't understand what is **Expression**. 

### ISSUE THREAD

**Issue by Quintri** [bpodgursky/jbool_expressions#44]
Hello, 

I am trying to minimize/simplify somes boolean expressions. 
I tried to lunch your project on my computer. I work with Eclipse but I had so many problems. I don't know why. For example, Eclipse doesen't understand what is **Expression**. 

Do you know why ? 
And also, to simplify your expressions, did you use boolean algebra simplification or Quine McCluskey method ? 

thanks you

**Comment by Quintri** [comment]
Taking the same example below, 

when I write 

`Expression<String> expr = ExprParser.parse("( ( (! C) | C) & A & B)")
`
Eclipse prefer 

`com.bpodgursky.jbool_expressions.Expression<String> expr = ExprParser.parse("( ( (! C) | C) & A & B)");
`

I have this problem : 

**java.lang.NoClassDefFoundError: org/antlr/runtime/RecognitionException**

thank you

**Comment by bpodgursky** [comment]
This project uses Maven to import dependencies, and uses antlr to generate the expression types (https://github.com/bpodgursky/jbool_expressions/blob/master/pom.xml#L133).  I don't use Eclipse personally, but I know it has support for importing projects from a Maven pom.xml; that will be the best way to make sure the types are available.

The normal simplification just uses boolean algebra but there is a method to use QMC as well -- see the comment here for more details https://github.com/bpodgursky/jbool_expressions/blob/master/src/main/java/com/bpodgursky/jbool_expressions/rules/RuleSet.java#L47

**Comment by zaza** [comment]
I just tried it myself, with Eclipse 4.22 and M2E (Maven integration for Eclipse), and it worked fine. I was able to import the project to my Eclipse workspace. It did ask me to install "m2e connector for antlr" while doing the import, but once it was done the project was compiled successfully and I even managed to run the snippets @Quintri posted above.

@bpodgursky I would close the ticket unless @Quintri objects.
